src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Icu.cs

```diff
@@ -833,6 +861,22 @@ private unsafe int IcuGetHashCodeOfString(ReadOnlySpan<char> source, CompareOpti
             Debug.Assert(!GlobalizationMode.UseNls);
             Debug.Assert((options & (CompareOptions.Ordinal | CompareOptions.OrdinalIgnoreCase)) == 0);
 
+#if TARGET_BROWSER
+            if (GlobalizationMode.Hybrid)
+            {
+                if (!_isInvariantCulture && !LocalizedHashCodeSupportsCompareOptions(options))
+                {
+                    throw new PlatformNotSupportedException(GetPNSEText("GetHashCode"));
+                }
+
+                // JS cannot create locale-sensitive HashCode, use invaraint functions instead
+                // empty chars influence invariant hashing algo but do not influence ICU hashing algo,
+                // Hybrid wants to behave like ICU
+                ReadOnlySpan<char> sanitizedSource = RemoveEmptyChars(source, options);
```

Comment:
I see that `GetHashCodeOrdinal` is heavily optimized, I guess because string hashing is very frequent operation.
Here we are allocating buffer and iterating over it and only then pass it to 
`InvariantGetHashCode` -> `GetHashCodeOrdinalIgnoreCase` -> `GetHashCodeOrdinal`. 

We could have loop over the original buffer which does all 3 things in one pass. For each character we could 
a) skip it 
b) lower-case it 
c) add it to the hash.

I'm not sure this optimization needs to be part of this PR.

Reply:
I would prefer another PR that would add tests to benchmark and based on the results, check if optimization is valid.